helper/ttl/validation.go

```diff
@@ -0,0 +1,83 @@
+package ttl
+
+import (
+	"errors"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/logical"
+)
+
+type MountHandler struct {
+	// ConfigTTL is the TTL being set at a mount's config level.
+	// For example, if your plugin were the Azure secrets engine,
+	// and you had a path for an overall config like <mount>/config,
+	// this would be the TTL at that level.
+	// Optional, can be unset.
+	ConfigTTL time.Duration
+
+	// ConfigMaxTTL is the MaxTTL being set at a mount's config level.
+	// For example, if your plugin were the Azure secrets engine,
+	// and you had a path for an overall config like <mount>/config,
+	// this would be the MaxTTL at that level.
+	// Optional, can be unset.
+	ConfigMaxTTL time.Duration
+
+	// RoleTTL is the TTL being set at a role level, which is a lower
+	// and more specific level than the config level.
+	// Optional, can be unset.
+	RoleTTL time.Duration
+
+	// RoleMaxTTL is the MaxTTL being set at a role level, which is a lower
+	// and more specific level than the config level.
+	// Optional, can be unset.
+	RoleMaxTTL time.Duration
+}
+
+func (h *MountHandler) Validate(system logical.SystemView) error {
```

Comment:
Maybe here you could take something narrower. Define an interface with just: `DefaultLeaseTTL() ` and `MaxLeaseTTL()` since that's all you use. From the consumer side, logical.SystemView would still be directly usable, but you'd avoid importing `logical`, which *might* avoid some circular imports.

Reply:
Hrm, circular import issues because of the logical package, good point. I'm not sure I totally grok what those methods would look like but I'll play with it a little.

Reply:
I just mean make something like this and have it be the type of the parameter into Validate:
```
type TTLView interface {
	DefaultLeaseTTL() time.Duration
	MaxLeaseTTL() time.Duration
}
```

This is just defining the subset of the `logical.SystemView` that we need. I'd still be able to call `foo.Validate(b.SystemView())` from my code but we'd decouple the helper from `logical`.